aten/src/TH/generic/THTensorLapack.c

```diff
@@ -101,6 +101,34 @@ static THTensor *THTensor_(cloneColumnMajor)(THTensor *self, THTensor *src)
   return THTensor_(cloneColumnMajorNrows)(self, src, src->size[0]);
 }
 
+/*
+ * Creates a clone of a 3D tensor such that each 2D batch is in column major
+ * format and each batch is contiguous WRT itself in memory.
+ */
+static THTensor *THTensor_(cloneBatchedColumnMajor)(THTensor *self, THTensor *src)
+{
+  THAssert(src->nDimension == 3);
+  int64_t matrix_size = src->size[1] * src->size[2];
+  if (self == src && self->stride[0] == matrix_size && 
+      self->stride[1] == 1 && self->stride[2] == self->size[1]) {
+    THTensor_(retain)(self);
+    return self;
+  }
+
+  if (self == src) {
+    self = THTensor_(new)();
+  } else {
+    THTensor_(retain)(self);
+  }
+
+  int64_t size[3] = { src->size[0], src->size[1], src->size[2] };
+  int64_t stride[3] = { matrix_size, 1, src->size[1] };
+
+  THTensor_(resizeNd)(self, 3, size, stride);
+  THTensor_(copy)(self, src);
+  return self;
+}
```

Comment:
I think it'd be better to rearrange some statements in this function. I'd move the `size` and `stride` arrays to the top and used that both to check if `self` matches what we expect, and in a call to `resizeNd` (less duplication, fewer chances for errors).